Runtime/src/closure-heap.ts

```diff
@@ -0,0 +1,22 @@
+import { ExportedFunctions } from "./types";
+
+/// Memory lifetime of closures in Swift are managed by Swift side
+export class SwiftClosureDeallocator {
+    private functionRegistry: FinalizationRegistry<number>;
+
+    constructor(exports: ExportedFunctions) {
+        if (typeof FinalizationRegistry === "undefined") {
+            throw new Error(
+                "The Swift part of JavaScriptKit was configured to require the availability of JavaScript WeakRefs. Please build with `-Xswiftc -DJAVASCRIPTKIT_WITHOUT_WEAKREFS` to disable features that use WeakRefs."
```

Comment:
nit: can we make this a multi-line string? No need to display it as multi-line to the user, just a matter of formatting in the source code.